fedireads/remote_user.py

```diff
@@ -25,37 +26,47 @@ def get_or_create_remote_user(actor):
         response.raise_for_status()
     data = response.json()
 
-    # the webfinger format for the username.
     actor_parts = urlparse(actor)
+    with transaction.atomic():
+        user = create_remote_user(data)
+        user.federated_server = get_or_create_remote_server(actor_parts.netloc)
+        user.save()
+
+    avatar = get_avatar(data)
+    user.avatar.save(*avatar)
+
+    if user.fedireads_user:
+        get_remote_reviews(user)
+    return user
+
+
+def create_remote_user(data):
+    ''' parse the activitypub actor data into a user '''
+    actor = data.get('id')
```

Comment:
Should just be:

```actor = data['id']```

(Otherwise, it fails with a hard to debug error trying to handle the none below.)

Reply:
Should we assert that the actor matches the actor we asked for? Otherwise the remote could write use the opportunity to impersonate someone.

(I'm trying to get the signature code to use this and accidentally testing interesting edge cases. :))

Reply:
hmm, I'm not sure what to check to verify the actor -- if it's loaded from the database, it's matching the actor string, and if it's created, it's set to whatever was sent. I think this is called after the signature would be verified?

Reply:
I think in get_or_create_remote_user, we need something like:

```
assert actor == data['id']
```

Otherwise we could fetch the user info for https://bad.example.com/cthulahoops and it could send us fake profile data for https://good.example.com/mouse. I think.

Reply:
oh, do you mean comparing the request url to the id provided in the activitypub json?